### ISSUE TITLE
recovery-target

### ISSUE THREAD

**Issue by glushakov** [postgrespro/pg_probackup#247]
Привет. 
В документации сказано:
Чтобы восстановить последнее возможное состояние, исходя из содержимого архива WAL, передайте в параметре --recovery-target значение latest:
однако это не совсем стыкуется со следующим параметром по умолчанию  (По умолчанию выбирается линия времени указанной резервной копии):
--recovery-target-timeline=current

Предположим, что я восстанавливался несколько раз, наплодив несколько новых timeline (допустим 7) 
Последний фулл бэкап у меня был в timeline=3.
Однако в timeline 7 у меня были изменения, которые отражены в WAL.
Если я запущу восстановление с ключем recvovery-target='latest' то получу не последнее состояние на timeline=7, а последнее из timeline 3.

Мне кажется, что это надо как отразить в документации, что восстановление происходит на последнее возможное состояние в рамках последнего бэкапа и его таймлайна.

**Comment by gsmol** [comment]
Добрый день.
Да, хорошее замечание.

**Comment by glushakov** [comment]
Хорошо иметь возможность восстанавливаться на последнее доступное состояние (или pitr) не вникая в таймлайны. Допустим мне надо восстановится на время 12-00, но это время у меня было в timeline=5. Мне при восстановлении тогда явно надо указать timeline=5. 
Было бы неплохо если probackup "знал" какое время восстановления доступно в каких таймлайнах и сам выбирал нужный

**Comment by gsmol** [comment]
Проблема в том, что понятие `последнее доступное состояние` довольно бессмысленно без таймлайнов.
И последнее не всегда находится в таймлайне с самым большим номером, т.е. мы не можем взять на себя такое допущение.
И
```
                /--------->t3
         /-------*>     t2   <- Последнее состояние
----A--*----B------> t1
```
Вы знаете, что t2 содержил последнее состояние, но pg_probackup никак не сможем это понять без подсказки от юзера.

> Было бы неплохо если probackup "знал" какое время восстановления доступно в каких таймлайнах и сам выбирал нужный
>
Мы думали об этом, но больше в контексте предоставления юзеру больше информации об архиве в `show --archive`.
Самим же за юзера выбирать таймлайн - это очень опасно.

**Comment by gsmol** [comment]
Да, у нас есть недоработка - юзер не может передать аргумент `latest` в `recovery-target-timeline`.
Если мы это добавим, это решит вашу проблему?

**Comment by glushakov** [comment]
> Да, у нас есть недоработка - юзер не может передать аргумент `latest` в `recovery-target-timeline`.
> Если мы это добавим, это решит вашу проблему?

Да это, кажется, поможет
Так же доработка "предоставления юзеру больше информации об архиве в show --archive" - было бы отлично!
Спасибо
